<launch>
    <node pkg="pointcloud_utils" type="ground_processor" name="ground_parser"
          required="true" output="screen" cwd="node">
         
        <!-- <param name="cloud_topic_in" 	value="/invz_topic_1" /> -->
        <param name="cloud_topic_in"  value="/velodyne_points" />      
		<param name="aligned_topic_out" value="/aligned_points" />
		<param name="ground_topic_out"  value="/ground_points" />
		<param name="nonground_topic_out" value="/nonground_points" />
        <param name="visualize_plane"   value="true" />

       	<!-- <param name="plane_x_min" value="3.0" />
        <param name="plane_x_max" value="4.0" />
        <param name="plane_y_min" value="-0.2" />
        <param name="plane_y_max" value="0.2" />
        <param name="plane_z_min" value="-10" />
        <param name="plane_z_max" value="10" /> -->

        <param name="plane_x_min" value="1" />
        <param name="plane_x_max" value="1.2" />
        <param name="plane_y_min" value="-1" />
        <param name="plane_y_max" value="0.0" />
        <param name="plane_z_min" value="0.0" />
        <param name="plane_z_max" value="0.5" /> 

		<param name="use_point_track_method" value="false" />
        
        <param name="track_plane_states_over_time" value="true" />
        <param name="iterate_plane_fit" value="false" />
        <param name="max_plane_fit_iterations" value="15" />
        <param name="outlier_point_tolerance" value="0.05" />
        <param name="min_points_to_fit" value="10" />

        <!-- <param name="iterate_plane_fit" value="true" />
        <param name="max_plane_fit_iterations" value="10" />
        <param name="outlier_point_tolerance" value="0.03" />
        <param name="min_points_to_fit" value="40" /> -->

        <param name="intensity_min"         value="0" /> #was 5
        <param name="intensity_max"         value="256" />

        <param name="report_offsets_at_origin" value="true" /> #the alternative is to report plane distance vector at origin
    

    	<param name="aligned_cloud_frame"	 value="/velodyne" />
    	<param name="ground_point_tolerance" value="0.05" />

    </node>
</launch>  